tests/soil_crop_tests/crop_tests/test_nutrient_uptake.py

```diff
@@ -0,0 +1,504 @@
+from math import log, exp
+
+import pytest
+from pytest_mock import MockerFixture
+
+from RUFAS.output_manager import OutputManager
+from RUFAS.routines.field.crop.crop_data import CropData
+from RUFAS.routines.field.crop.nutrient_uptake import NutrientUptake
+
+
+@pytest.mark.parametrize(
+    "halfheat,heatfrac,emerge,half,near,mature,should_fail",
+    [
+        (0.5, 1.0, 0.8, 0.6, 0.3, 0.2, False),  # start
+        (0.99, 1.0, 0.8, 0.6, 0.3, 0.2, False),  # half_heat close to mature heat
+        (0.01, 1.0, 0.8, 0.6, 0.3, 0.2, False),  # small half_heat
+        (0.5, 1.0, 0.8, 0.6, 0.20001, 0.2, False),  # near very close to mature
+        (0.286, 0.54, 0.522, 0.4, 0.1, 0.08, False),  # arbitrary
+        # Above tests are copied from old subroutine tests
+        (0.8, 1, 0.9, 0.6, 0.3, 0.25, False),
+        (0.8, 0.81, 0.9, 0.6, 0.3, 0.25, False),  # small difference in heat units
+        (
+            0.8,
+            1,
+            0.9,
+            0.6,
+            0.25000001,
+            0.25,
+            False,
+        ),  # small difference in nfrac_near and nfrac_3
+        (0.633, 0.691, 0.530, 0.101, 0.057, 0.013, False),  # arbitrary
+        (0.5, 0.5, 0.530, 0.101, 0.057, 0.013, True),
+    ],
+)
+def test_determine_nutrient_shape_parameters(
+    halfheat: float,
+    heatfrac: float,
+    emerge: float,
+    half: float,
+    near: float,
+    mature: float,
+    should_fail: bool,
+    mocker: MockerFixture,
+) -> None:
+    """check that the shape parameters are correctly calculated by determine_nshapes() and that errors were raised
+    correctly"""
```

Comment:
```suggestion
    """Check that the shape parameters are correctly calculated by determine_nshapes() and that errors were raised
    correctly."""
```
Please make same changes to all docstrings in this class.